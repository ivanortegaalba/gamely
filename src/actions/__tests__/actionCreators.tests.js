import { addToFavouritesUnsafe, fetchGames, filterGamesBy, removeFromFavouritesUnsafe
} from '../actionCreators'
import gameMock from '../../__mocks__/game'

describe('actionCreators', () => {
  describe('fetchGames', () => {
    it('should return a call api action', () => {
      expect(fetchGames()).toMatchSnapshot()
    })
  })

  describe('addToFavouritesUnsafe', () => {
    it('should return an action with a the game short name', () => {
      expect(addToFavouritesUnsafe(gameMock.short)).toMatchSnapshot()
    })
  })

  describe('removeFromFavouritesUnsafe', () => {
    it('should return an action with a the game short name', () => {
      expect(removeFromFavouritesUnsafe(gameMock.short)).toMatchSnapshot()
    })
  })

  describe('filterGamesBy', () => {
    it('should return an action with the value which will be used to filter', () => {
      expect(filterGamesBy(gameMock.name)).toMatchSnapshot()
    })
  })
})
