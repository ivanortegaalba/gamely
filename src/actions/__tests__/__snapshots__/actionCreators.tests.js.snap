// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[
  `actionCreators addToFavouritesUnsafe should return an action with a the game short name 1`
] = `
Object {
  "shortName": "eightballpool",
  "type": "ADD_TO_FAVOURITES",
}
`;

exports[`actionCreators fetchGames should return a call api action 1`] = `
Object {
  "Call API": Object {
    "endpoint": "/resources/games.json",
    "schema": ArraySchema {
      "schema": EntitySchema {
        "_getId": [Function],
        "_idAttribute": "short",
        "_key": "games",
        "_mergeStrategy": [Function],
        "_processStrategy": [Function],
        "schema": Object {},
      },
    },
    "types": Array [
      "GAMES_REQUEST",
      "GAMES_SUCCESS",
      "GAMES_FAILURE",
    ],
  },
}
`;

exports[
  `actionCreators filterGamesBy should return an action with the value which will be used to filter 1`
] = `
Object {
  "filterBy": "8-Ball Pool",
  "type": "FILTER_BY",
}
`;

exports[
  `actionCreators removeFromFavouritesUnsafe should return an action with a the game short name 1`
] = `
Object {
  "shortName": "eightballpool",
  "type": "REMOVE_FROM_FAVOURITES",
}
`;
